index.js

```diff
@@ -131,11 +137,12 @@ module.exports = {
       // compare i18n number in req and in global
       // if they don't match, it means a language had a translation piece edited
       // so need to reload this i18n language file
-      if (options.autoReload && self.lastI18nGeneration !== req.data.global.i18nGeneration) {
-        const locale = self.getLocale(req);
+      const locale = self.getLocale(req);
+      self.lastI18nGeneration = self.lastI18nGeneration || {};
+      if (options.autoReload && self.lastI18nGeneration[locale] !== req.data.global.i18nGeneration[locale]) {
         await saveI18nFile({ locale, ...options });
       }
-      self.lastI18nGeneration = req.data.global.i18nGeneration;
+      self.lastI18nGeneration[locale] = req.data.global.i18nGeneration[locale];
```

Comment:
I am not convinced this is the solution (see above). Testing is required to reproduce the issue.

Reply:
I think the problem comes from 
```
self.lastI18nGeneration = req.data.global.i18nGeneration;
```

When doing this, the latest locale is copied to `self.lastI18nGeneration`, but not all locales that have been visited. So, the next visit on a previously visited locale will trigger a file reload.

Reply:
Oh. Yes. That is a bug. We can't have one global `lastI18nGeneration` value since locales evolve at different rates.

However we also do not need to create an object inside `req.data.global.i18nGeneration`. We just need to copy that value, which is already locale-specific, to the correct sub-property of `self.lastI18nGeneration`.

So `self.lastI18nGeneration[locale]` is the right idea, but `req.data.global.i18nGeneration[locale]` is redundant.